src/rust/engine/rule_graph/src/rules.rs

```diff
@@ -46,4 +46,15 @@ pub trait Rule: Clone + Debug + Display + Hash + Eq + Sized + 'static {
   /// True if this rule implementation should be required to be reachable in the RuleGraph.
   ///
   fn require_reachable(&self) -> bool;
+
+  ///
+  /// True if this rule implementation is provided by Rust code, not Python.
+  ///
+  // TODO: Convert this into some sort of enum!
+  fn is_intrinsic(&self) -> bool;
```

Comment:
This is probably not a concept that should be exposed in this interface, as it is specific to the engine's rule graph. Is it possible to either drop it, or incorporate it into the `fmt_for_graph` display...?

Reply:
I've changed this to be named `color()` and return an `Option<Palette>` for any special color the rule should be painted as! This removes any knowledge of "intrinsics" from the `rule_graph` crate!